signalfx-java/src/main/java/com/signalfx/signalflow/StreamMessage.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2016 SignalFx, Inc. All rights reserved.
+ */
+package com.signalfx.signalflow;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Base class for stream messages received from a SignalFlow computation
+ * 
+ * @author dgriff
+ */
+public class StreamMessage {
+
+    /**
+     * Enumeration of kinds of stream messages
+     */
+    public static enum Kind {
+
+        CONTROL("control-message"),
+        INFORMATION("message"),
+        METADATA("metadata"),
+        DATA("data"),
+        EVENT("event"),
+        ERROR("error");
+
+        private final String specName;
+
+        Kind(String specName) {
+            this.specName = specName;
+        }
+
+        public String toString() {
+            return this.specName;
+        }
+    };
+
+    private static Map<String, Kind> SPEC_NAME_KINDS = new HashMap<String, Kind>();
+    static {
```

Comment:
That's... almost what I meant, but not quite. Just add this inside the `Kind` enum above after the enum entries:

```java
private static final Map<String, Kind> KINDS = new HashMap<>();
static {
    for (Kind kind : Kind.values()) {
        KINDS.put(kind.specName, kind);
    }
}

public static Kind fromSpecName(String specName) {
    Kind kind = KINDS.get(specName);
    Preconditions.checkArgument(kind != null);
    return kind;
}
```